ckan/logic/action/get.py

```diff
@@ -1875,8 +1876,12 @@ def package_search(context, data_dict):
         else:
             data_source = 'validated_data_dict'
         data_dict.pop('use_default_schema', None)
-        # return a list of package ids
-        data_dict['fl'] = 'id {0}'.format(data_source)
+
+        result_fl = data_dict.get('fl')
+        if not result_fl:
+            data_dict['fl'] = 'id {0}'.format(data_source)
+        else:
+            data_dict['fl'] = ' '.join(result_fl)
```

Comment:
the solr API should accept a real list, doesn't it? converting it ourselves will cause strange behaviour if we generate something that doesn't parse as a real list.